src/blog/jspi-newapi.md

```diff
@@ -0,0 +1,106 @@
+
+This change will likely have minimal impact on most applications; however, some applications will see a notable benefit by avoiding unnecessary trips to the browser event loop.
+
+### The new API
+
+The API is pretty simple: there is a function that takes a function exported from a WebAssembly module and converts it into a `Promise` returning function:
+
+```js
+Function Webassembly.promising(Function wsFun)
+```
+
+Note that, although the argument is typed as a `Function` &mdash; i.e., a JavaScript function &mdash; it is actually restricted to WebAssembly functions.
+
+On the suspending side, we have a new class `WebAssembly.Suspending`, together with a constructor that takes a JavaScript function as argument. In WebIDL this is written:
+
+```js
+interface Suspending{
+    constructor (Function fun);
+}
+```
+
+Note that this API has an asymmetric feel to it: we have a function that takes a WebAssembly function and returns a new promising (sic) function; whereas to mark a suspending function we enclose it in a `Suspending` object. This reflects a deeper reality about what is happening under the hood.
+
+The suspending behavior of an import is intrinsically part of the *call* to the import: i.e., some function inside the instantiated module calls the import and suspends as a result.
+
+On the other hand, the `promising` function takes a regular WebAssembly function and returns a new one that can respond to being suspended and which returns a `Promise`.
+
+### Using the new API
+
+If you are an Emscripten user, then using the new API will typically involve no changes to your code. You must be using a version of emscripten that is at least 3.1.61, and you must be using a version of Chrome that is at least 126.0.6478.17 (milestone 126).
+
+If you are rolling your own integration, then your code should be significantly simpler. In particular, it is no longer necessary to have code that stores the passed-in `Suspender` object (and retrieve it when calling the import). One can simply use regular sequential code within the WebAssembly module.
+
+### The old API
+
+The old API will continue to operate at least until 10/29/2024 (Chrome M128). After that we plan on removing the old API.
+
+Note that Emscripten itself will no longer support the old API as of version 3.1.61.
+
+## What is happening with JSPI?
+
+### Implementation aspects
+
+The biggest change to JSPI that we are working on is actually invisible to most programmers: so-called growable stacks.
+
+The current implementation of JSPI is based on allocating stacks of a fixed size. In fact, the allocated stacks are rather large; because we have to be able to accomodate arbitrary WebAssembly computations which may require deep stacks to handle recursion properly.
+
+However, this is not a sustainable strategy: we would like to support applications with millions of suspended coroutines; this is not possible if each stack is 1Mb in size.
```

Comment:
Did you mean Megabyte (MB) or Megabit (Mb) here? Just to make sure it's not a typo.

Reply:
It's a megabyte.

Reply:
MB then. Committed this change.